src/exo/cursors.py

```diff
@@ -0,0 +1,228 @@
+from __future__ import annotations
+
+import weakref
+from dataclasses import dataclass
+from enum import Enum
+from typing import Union, List, Iterable
+from weakref import ReferenceType
+
+from .API_types import ProcedureBase
+from .LoopIR import LoopIR
+
+
+class CursorKind(Enum):
+    Node = 0  # Pointing directly at a node
+    # BlockFront = 1  # Gap before the first statement of a block
+    # BlockEnd = 2  # Gap after the last statement of a block
```

Comment:
Are sequences of statements not supported?

Reply:
A sequence of statements is currently represented as a list of Cursor objects.